syntax = "proto3"; // 声明protobuf的版本

package fixbug; // 声明所在的包（对于C++来说就是namespace）

option cc_generic_services = true; // 定义选项，标识生成service服务类和rpc方法描述，默认不生成。

//自定义 返回码 消息类型
message ResultCode {
    int32 errcode = 1;
    bytes errmsg = 2;
}

/**/
//自定义 获取列表请求 消息类型
message GetFriendsListRequest {
    uint32 userid = 1;

}

//自定义 获取列表响应 消息类型
message GetFriendsListResponse {
    ResultCode result = 1; //返回码
    repeated bytes friends = 2; //名称列表
}



//自定义服务 好友服务的rpc框架
service FriendServiceRpc {
    //参数：登录请求    返回值：登录回应   
    rpc GetFriendsList(GetFriendsListRequest) returns(GetFriendsListResponse);
}
